plugins/workspace-minimap/src/focus_mode.ts

```diff
@@ -0,0 +1,171 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview A positionable version of the minimap.
+ * @author cesarades@google.com (Cesar Ades)
+ */
+
+import * as Blockly from 'blockly/core';
+
+const BlockEvents = new Set([
+  Blockly.Events.VIEWPORT_CHANGE,
+  Blockly.Events.BLOCK_CHANGE,
+  Blockly.Events.BLOCK_CREATE,
+  Blockly.Events.BLOCK_DELETE,
+  Blockly.Events.BLOCK_DRAG,
+  Blockly.Events.BLOCK_MOVE]);
+
+/**
+ * A class that highlights the user's viewport on the minimap.
+ */
+export class FocusMode {
+    private primaryWorkspace: Blockly.WorkspaceSvg;
+    private minimapWorkspace: Blockly.WorkspaceSvg;
+    private onChangeWrapper: (e: Blockly.Events.Abstract) => void;
+    private svgGroup: SVGElement;
+    private rect: SVGElement;
+    private background: SVGElement;
+    private id: string;
+
+
+    /**
+     * Constructor for focus mode.
+     * @param primary The primary workspaceSvg.
+     * @param minimap The minimap workspaceSvg.
+     */
+    constructor(primary: Blockly.WorkspaceSvg, minimap: Blockly.WorkspaceSvg) {
+      this.primaryWorkspace = primary;
+      this.minimapWorkspace = minimap;
```

Comment:
You can also use parameter properties for these! https://medium.com/@daveford/parameter-properties-in-typescript-c99df6ba13f7